quinn-proto/src/connection/mod.rs

```diff
@@ -904,6 +910,9 @@ impl Connection {
             if pad_datagram {
                 builder.pad_to(MIN_INITIAL_SIZE);
             }
+            if pad_datagram_to_mtu {
+                builder.pad_to(self.path.current_mtu());
```

Comment:
I think this is incorrect for tail loss probes, which must never exceed `INITIAL_MTU` so that losses due to an MTU black hole can be detected and handled. Instead, we should pad to whatever the size limit is for the current datagram (see e.g. `PacketBuilder::max_size`, but note this is an absolute offset in the GSO batch).

Reply:
Yes, you are right. This was incorrect for tail loss probes. Same issue in the GSO batch processing.

It took me some time to fully understand the `poll_transmit` logic, but it's fixed now. I've update the PR to only introduce one additional `builder.pad_to()`, which is protected by the loss probe check now and . The regular GSO batch padding can take care of the rest with a small modification.

The i686-linux-android failure is new and odd. I don't see how any change in behavior is possible with `self.config.pad_to_mtu == false`.